src/Adapter/LegacyContext.php

```diff
@@ -343,4 +345,13 @@ public function getAvailableLanguages()
     {
         return $this->getLanguages(false);
     }
+
+    /**
+     * @param Context $testInstance
+     *                              Unit testing purpose only
+     */
+    public static function setInstanceForTesting(Context $testInstance)
```

Comment:
???? Having a method only for testing purpose is not common at all

Reply:
He's right, setInstance should be enough. If you need it in your test, you probably did something wrong, or make it available for everyone :)

```suggestion
    public static function setInstance(Context $testInstance)
```

Reply:
That permits to keep consistance between another classes

![image](https://user-images.githubusercontent.com/1533248/108847934-a8cc0880-75e0-11eb-91cf-02d5f0ba92ac.png)

Reply:
Question is, shall we move in direction when we code to meet wrong standards, or fix them when we have possibility?